### ISSUE TITLE
publib-maven fails during `sign-and-deploy-file` step with PINENTRY_LAUNCHED 2145 curses 1.1.1 not a tty

### ISSUE THREAD

**Issue by AgustinBettati** [cdklabs/publib#1107]
### Command being run
```npx -p publib@latest publib-maven```

Uses version `publib@0.2.802`

### Env variables provided

```
env:
          MAVEN_GPG_PRIVATE_KEY: ${{ secrets.MAVEN_GPG_PRIVATE_KEY }}
          MAVEN_GPG_PRIVATE_KEY_PASSPHRASE: ${{ secrets.MAVEN_GPG_PRIVATE_KEY_PASSPHRASE }}
          MAVEN_PASSWORD: ${{ secrets.MAVEN_PASSWORD }}
          MAVEN_USERNAME: ${{ secrets.MAVEN_USERNAME }}
          MAVEN_STAGING_PROFILE_ID: ${{ secrets.MAVEN_STAGING_PROFILE_ID }}
```

### Complete error message

```
[INFO] Signer 'gpg' is signing 4 files
[DEBUG] 2.2.27
[GNUPG:] KEY_CONSIDERED 0E2DB99D44DEA2E3BE4A9D3326689A5EA6A5AFBF 2
[GNUPG:] BEGIN_SIGNING H10
[GNUPG:] PINENTRY_LAUNCHED 2145 curses 1.1.1 not a tty - - ? 1001/127 -
gpg: signing failed: No such file or directory
[GNUPG:] FAILURE sign 83918929
gpg: signing failed: No such file or directory
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.226 s
[INFO] Finished at: 2024-03-12T10:44:01Z
[INFO] ------------------------------------------------------------------------
Error:  Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:3.2.0:sign-and-deploy-file (default-cli) on project standalone-pom: Exit code: 2 -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:3.2.0:sign-and-deploy-file (default-cli) on project standalone-pom: Exit code: 2
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:375)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:351)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:171)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:163)
```

### Additional details

This command is being run with a GitHub workflow: https://github.com/mongodb/awscdk-resources-mongodbatlas/blob/main/.github/workflows/release.yml#L120. 
This publish workflow was working without issues in the past, with the last successful release on the 6th of February. I can see that a new version of [maven-gpg-plugin (3.2.0)](https://maven.apache.org/plugins-archives/) was release on the 8th of March, not sure if this had an impact.

When running the publish command locally (on macOS) I was requested to enter the GPG passphrase even when having the `MAVEN_GPG_PRIVATE_KEY_PASSPHRASE` env variable defined.

Any inputs or possible workarounds would be greatly appreciated.

**Comment by agdimech** [comment]
I too am seeing this issue!

**Comment by rix0rrr** [comment]
It seems this was a bug, fixed in `maven-gpg-plugin` release 3.2.1: https://github.com/apache/maven-gpg-plugin/releases/tag/maven-gpg-plugin-3.2.1

**Comment by rix0rrr** [comment]
Given that there are no more comments on this issue, I'm assuming this did indeed fix the issue. Closing.
